File: app/src/main/java/com/kaciula/archiman/presentation/screen/home/HomeContract.java
package com.kaciula.archiman.presentation.screen.home;

import com.kaciula.archiman.presentation.util.BasePresenter;

public interface HomeContract {

  interface View {

    void render(HomeViewModel viewModel);

    void goToUserDetailsScreen(UserViewModel user);
  }


  interface Presenter extends BasePresenter {

    void onClickRetry();

    void onClickUser(UserViewModel user);

    void onClickOkUserDialog(UserViewModel user);

    void onCancelUserDialog();

    void onClickUserDetails(UserViewModel user);
  }
}

File: app/src/main/kotlin/com/kaciula/archiman/presentation/screen/home/HomeContract.kt
package com.kaciula.archiman.presentation.screen.home

import com.kaciula.archiman.presentation.util.BasePresenter
import com.kaciula.archiman.presentation.util.ElmPresenter

interface HomeContract {

    interface View {

        fun render(state: HomeState)

        fun goToUserDetailsScreen(user: UserViewModel)
    }


    interface Presenter : BasePresenter, ElmPresenter {

        fun onDoneInitialize()

        fun onClickRetry()

        fun onClickUser(user: UserViewModel)

        fun onShowingUserDialog()

        fun onClickOkUserDialog(user: UserViewModel)

        fun onCancelUserDialog()

        fun onClickUserDetails(user: UserViewModel)
    }
}
